### PR TITLE
chain+wallet: make use of the new errors from `rpcclient`

### PR DESCRIPTION
**Pull request by yyforyongyu**
This PR removes the string matching and makes use of the newly defined errors inside `rpcclient`.

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -3760,27 +3741,23 @@ func (w *Wallet) publishTransaction(tx *wire.MsgTx) (*chainhash.Hash, error) {
...
 	_, err = chainClient.SendRawTransaction(tx, false)
 	if err == nil {
 		return &txid, nil
 	}
 
-	returnErr = MapBroadcastBackendError(err)
-
-	var errInMempool *ErrInMempool
-	var errAlreadyConfirmed *ErrAlreadyConfirmed
+	// Map the error to an RPC-specific error type.
```
Comment (ziggie1984) [inline wallet/wallet.go:3750]: Nit: Maybe a more detailed comment that all errors are mapped to bitcoind errors no matter the used backend

**Review by ziggie1984 — APPROVED** [review]
LGTM thank you for the cleanup makes definitely sense to put the error mapping into the btcd package 🙏

**Review by guggero — APPROVED** [review]
Very nice! LGTM pending existing comments (and dependent PR).

File: wallet/wallet.go (hunk 1 above)
Comment (yyforyongyu) [reply wallet/wallet.go:3750]: updated!

**Comment by guggero** [comment]
Can be updated with final version of `btcd`.

**Pull request merged**